clients/go/gengo.go

```diff
@@ -72,38 +85,196 @@ type WagClient struct {
 	// Keep the circuit doer around so that we can turn it on / off
 	circuitDoer    *circuitBreakerDoer
 	defaultTimeout time.Duration
-	logger       logger.KayveeLogger
+	logger      waglogger.WagClientLogger
 }
 
 var _ Client = (*WagClient)(nil)
 
+
+//This pattern is used instead of using closures for greater transparency and the ability to implement additional interfaces.
+type options struct {
+	transport    http.RoundTripper
+	logger       waglogger.WagClientLogger
+	instrumentor Instrumentor
+	exporter     sdktrace.SpanExporter
+}
+
+type Option interface {
+	apply(*options)
+}
+
+//Logger
```

Comment:
nit: this comment (and others like it that aren't attached to any code) don't really add to my understanding of the file and I think they can be removed

Reply:
Gotcha, I'll remove them later, they were mostly for MY understanding/organization of the code.